impls/src/adapters/qr.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2019 The Epic Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//use image::{ImageBuffer, Rgb};
+use super::compress::{compress, decompress, CompressionFormat};
+/// File Output 'plugin' implementation
+use crate::libwallet::{Error, ErrorKind, Slate, SlateVersion, VersionedSlate};
+use crate::{SlateGetter, SlatePutter};
+use image::Luma;
+use qrcode::QrCode;
+use quircs;
+use std::path::PathBuf;
+
+/// The default version of the QR in this version of Epic
+const QR_VERSION: u8 = 1;
+
+const LIMIT_QR_BIN: f32 = 2330.0;
+
+pub const RESPONSE_EXTENTION: &str = "response_";
+
+/// Saves all information and the type of compressor and which version of the epic for the emoji we are using
+struct QrHeader {
+	/// Method to compress/decompress
+	algo: CompressionFormat,
+	/// Version of emoji in Epic
+	version: u8,
+}
```

Comment:
This struct shouldn't be the same for QR and Emoji?

Reply:
Exactly, we have a header added in the first element of the Vec<u8>, it controls which compression algorithm to use